src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -541,7 +562,11 @@ private static IEnumerable<FunctionOverload> GetSystemOverloads(IFeatureProvider
                 .WithRequiredParameter("predicate", TwoParamLambda(LanguageConstants.Any, LanguageConstants.Any, LanguageConstants.Bool), "The predicate used to compare two array elements for ordering. If true, the second element will be ordered after the first in the output array.",
                     calculator: getArgumentType => CalculateLambdaFromArrayParam(getArgumentType, 0, t => TwoParamLambda(t, t, LanguageConstants.Bool)))
                 .WithReturnResultBuilder((binder, fileResolver, diagnostics, arguments, argumentTypes) => {
-                    return new(argumentTypes[0]);
+                    return new(argumentTypes[0] switch
+                    {
+                        TupleType tuple => new TypedArrayType(tuple.Item, TypeSymbolValidationFlags.Default),
```

Comment:
Worth creating a reusable function to do this? (Primarily for the sake of having a place to explain why we're doing this in a comment!)